From b04d8d4a02fbf070a017c11754501a27e8f99603 Mon Sep 17 00:00:00 2001
From: Ajay Kumar Gupta <ajay.gupta@ti.com>
Date: Tue, 20 Jan 2009 18:52:52 +0530
Subject: [PATCH 192/193] 024-ehci-fix-for-silicon-rev-changes.patch

---
 drivers/usb/host/ehci-omap.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/host/ehci-omap.c b/drivers/usb/host/ehci-omap.c
index a9e3d78..7cd3e0e 100644
--- a/drivers/usb/host/ehci-omap.c
+++ b/drivers/usb/host/ehci-omap.c
@@ -324,7 +324,7 @@ static int omap_start_ehc(struct platform_device *dev, struct usb_hcd *hcd)
  *
  */
 
-	if (omap_rev() > OMAP3430_REV_ES2_1) {
+	if (omap_silicon_rev() > OMAP35XX_MASK_ES2_1) {
 
 #ifndef CONFIG_OMAP_EHCI_PHY_MODE_PORT1
 		uhh_hostconfig_value |=
@@ -348,7 +348,7 @@ static int omap_start_ehc(struct platform_device *dev, struct usb_hcd *hcd)
 
 	omap_writel(uhh_hostconfig_value, OMAP_UHH_HOSTCONFIG);
 
-	if (omap_rev() < OMAP3430_REV_ES3_0) {
+	if (omap_silicon_rev() < OMAP35XX_MASK_ES3_0) {
 		/* Ensure that BYPASS is set */
 		while (omap_readl(OMAP_UHH_HOSTCONFIG)
 				& (1 << OMAP_UHH_HOSTCONFIG_ULPI_BYPASS_SHIFT))
-- 
1.5.6

