From a20b52f2c8d677e4fe64c0240288997053e3cc5a Mon Sep 17 00:00:00 2001
From: Sergio Aguirre <saaguirre@ti.com>
Date: Mon, 26 Jan 2009 15:19:28 -0600
Subject: [PATCH 259/364] omap34xxcam: Get rid of hw resources

Signed-off-by: Sakari Ailus <sakari.ailus@nokia.com>
Signed-off-by: Sergio Aguirre <saaguirre@ti.com>
---
 drivers/media/video/omap34xxcam.h |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/drivers/media/video/omap34xxcam.h b/drivers/media/video/omap34xxcam.h
index ffb5696..78458e1 100644
--- a/drivers/media/video/omap34xxcam.h
+++ b/drivers/media/video/omap34xxcam.h
@@ -168,10 +168,6 @@ struct omap34xxcam_videodev {
  * protected by the lock above.
  * @sgdma: ISP sgdma subsystem information structure
  * @dma_notify: DMA notify flag
- * @irq: irq number platform HW resource
- * @mmio_base: register map memory base (platform HW resource)
- * @mmio_base_phys: register map memory base physical address
- * @mmio_size: register map memory size
  * @dev: device structure
  * @vdevs: /dev/video specific structures
  * @fck: camera module fck clock information
-- 
1.6.0.3

