From 43519f6da7d0507ba213ffe0bb897ee8dbf9decd Mon Sep 17 00:00:00 2001
From: Ajay Kumar Gupta <ajay.gupta@ti.com>
Date: Thu, 5 Feb 2009 17:10:49 +0530
Subject: [PATCH 016/364] usb: ehci: fix ehci issue when built as module

---
 drivers/usb/host/ehci-omap.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/usb/host/ehci-omap.c b/drivers/usb/host/ehci-omap.c
index 6152f18..76d9f1b 100644
--- a/drivers/usb/host/ehci-omap.c
+++ b/drivers/usb/host/ehci-omap.c
@@ -179,11 +179,13 @@ static int omap_start_ehc(struct platform_device *dev, struct usb_hcd *hcd)
 			(7 << OMAP3430ES2_EN_PERIPH2_DPLL_SHIFT),
 			PLL_MOD, OMAP3430ES2_CM_CLKEN2);
 
+#if 0 /* initialization stucks here when EHCI built as module */
 	while (!(cm_read_mod_reg(PLL_MOD, CM_IDLEST2) &
 				OMAP3430ES2_ST_PERIPH2_CLK_MASK))
 		dev_dbg(hcd->self.controller,
 			"idlest2 = 0x%x\n",
 			cm_read_mod_reg(PLL_MOD, CM_IDLEST2));
+#endif
 	/* End DPLL5 programming */
 
 
-- 
1.5.6

