From f73e07def5ea0e6f9d2dbf600adef6e159045798 Mon Sep 17 00:00:00 2001
From: Sergio Aguirre <saaguirre@ti.com>
Date: Mon, 26 Jan 2009 15:19:34 -0600
Subject: [PATCH 246/364] OMAP3ISP: Core: Remove idle mode settings from mmu

Signed-off-by: Sakari Ailus <sakari.ailus@nokia.com>
Signed-off-by: Sergio Aguirre <saaguirre@ti.com>
---
 drivers/media/video/isp/ispmmu.c |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/drivers/media/video/isp/ispmmu.c b/drivers/media/video/isp/ispmmu.c
index eb7f6c6..076aea1 100644
--- a/drivers/media/video/isp/ispmmu.c
+++ b/drivers/media/video/isp/ispmmu.c
@@ -564,7 +564,6 @@ static void ispmmu_isr(unsigned long status, isp_vbq_callback_ptr arg1,
 int __init ispmmu_init(void)
 {
 	int i, val = 0;
-	struct isp_sysc isp_sysconfig;
 
 	isp_get();
 
@@ -577,9 +576,6 @@ int __init ispmmu_init(void)
 		isp_put();
 		return -ENODEV;
 	}
-	isp_sysconfig.reset = 0;
-	isp_sysconfig.idle_mode = 1;
-	isp_power_settings(isp_sysconfig);
 
 	ttb_page = alloc_pages(GFP_KERNEL, get_order(ISPMMU_TTB_ENTRIES_NR *
 									4));
-- 
1.5.6

