From de420351d6de57c7e592f4c3305fa020c57e4299 Mon Sep 17 00:00:00 2001
From: Sergio Aguirre <saaguirre@ti.com>
Date: Mon, 26 Jan 2009 15:19:30 -0600
Subject: [PATCH 256/364] omap34xxcam: Power down slaves at streamoff unless videobuf_streamoff fails

Signed-off-by: Sakari Ailus <sakari.ailus@nokia.com>
Signed-off-by: Sergio Aguirre <saaguirre@ti.com>
---
 drivers/media/video/omap34xxcam.c |   11 ++++++-----
 1 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/media/video/omap34xxcam.c b/drivers/media/video/omap34xxcam.c
index e0b507f..47d2e94 100644
--- a/drivers/media/video/omap34xxcam.c
+++ b/drivers/media/video/omap34xxcam.c
@@ -826,13 +826,14 @@ static int vidioc_streamoff(struct file *file, void *fh, enum v4l2_buf_type i)
 		isp_stop();
 
 	rval = videobuf_streamoff(q);
-	if (!rval)
+	if (!rval) {
 		vdev->streaming = NULL;
 
-	omap34xxcam_slave_power_set(vdev, V4L2_POWER_STANDBY,
-					OMAP34XXCAM_SLAVE_POWER_SENSOR);
-	omap34xxcam_slave_power_suggest(vdev, V4L2_POWER_STANDBY,
-					OMAP34XXCAM_SLAVE_POWER_LENS);
+		omap34xxcam_slave_power_set(vdev, V4L2_POWER_STANDBY,
+					    OMAP34XXCAM_SLAVE_POWER_SENSOR);
+		omap34xxcam_slave_power_suggest(vdev, V4L2_POWER_STANDBY,
+						OMAP34XXCAM_SLAVE_POWER_LENS);
+	}
 
 	mutex_unlock(&vdev->mutex);
 
-- 
1.5.6

